es-space-mgmt/main/default/lwc/reservationHelperForm/__tests__/reservationHelperForm.test.js

```diff
@@ -1,6 +1,26 @@
 import { createElement } from 'lwc';
 import { FlowNavigationNextEventName } from 'lightning/flowSupport';
 import ReservationHelperForm from 'c/reservationHelperForm';
+import getCustomerFields from '@salesforce/apex/customerServices.getCustomerFields';
+
+// Mock realistic data
+const APEX_GET_CUSTOMER_FIELDS_SUCCESS = {
+    city: 'MailingCity',
+    email: 'Email',
+    name: 'Name',
+    state: 'MailingState'
+};
+
+// Mocking imperative Apex method call
+jest.mock(
+    '@salesforce/apex/customerServices.getCustomerFields',
+    () => {
+        return {
+            default: jest.fn()
+        };
+    },
+    { virtual: true }
+);
```

Comment:
@pozil I had to add this in since after I started using the `0.13` version of `sfdx-lwc-jest`, for the component that has a child component involving imperative apex, I  received the undefined error in jest tests whenever imperative apex call was made. Not sure if this is a feature or a bug.

I had to manually add the mocks for it to pass.

Reply:
Thanks for taking care of this @msrivastav13.
@jodarove what's your take on this, is that expected behavior?

Reply:
@pozil, i don't think this is the expected behavior; @msrivastav13, (if possible) let me take a look before you merge so I can determine why this apex method is not being auto-mocked.

Reply:
Thanks, @jodarove! Looking forward to seeing what you find!

Reply:
created https://github.com/salesforce/sfdx-lwc-jest/issues/230

Reply:
@pozil Just FYI, there will be a new version once this bug is patched, and will have to upgrade all the sample apps with the newer version.

Also looks like now the imperative apex calls from LWC will be auto-mocked, so might be some more work to incorporate them as well.

Reply:
@msrivastav13, @pozil the issue is fixed (v0.12.6 (summer21) v0.14.0 (next). however, we had to remove the auto-mocks for apex and apexContinuation modules.

Reply:
@jodarove too bad we couldn't keep it. I understand that we have to revert to the using `registerApexTestWireAdapter` however, I'm seeing a deprecation warning if I do that in v0.12.6 or v0.14.0:
```
console.warn
    registerApexTestWireAdapter is deprecated. More details: https://github.com/salesforce/wire-service-jest-util/blob/master/docs/migrating-from-version-2.x-to-3.x.md
```

Did I miss something?

Reply:
@ jodarove Please note that `0.14.0` breaks the tests for the wire adapter change.  I was hoping I have to only auto mock for the imperative apex and not for the wire adapters. Is it a bug or I am missing something?

Reply:
@pozil, I added a note on the [migration guide](https://github.com/salesforce/wire-service-jest-util/blob/master/docs/migrating-from-version-2.x-to-3.x.md#faq). Is not that we reverted to `registerApexTestWireAdapter` (which is still deprecated), is that the platform no longer auto-mocks apex methods (to a valid ApexTestWireAdapter). In order to remove the `registerApexTestWireAdapter` calls, test authors will need to mock their apex methods. cc/ @pmdartus

Reply:
@msrivastav13 can you elaborate more on breaks "wire adapter change" (check [this comment](https://github.com/trailheadapps/easy-spaces-lwc/pull/257#discussion_r662439382)).

> I was hoping I have to only auto mock for the imperative apex and not for the wire adapters

When you mock "imperative apex" you also mock the "wire adapters"; they are both the same module.

Reply:
ah! ok, I was assuming that for `@wire` decorated apex I do not need to do anything and it is auto-mocked which it seems is not possible without actually mocking it.